membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -38,56 +39,53 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
         })
       )
     } yield {
-      AttributesMaker.attributes(identityId, subscriptions, forDate)
+      AttributesMaker.zuoraAttributes(identityId, subscriptions, forDate)
     }
 
-
-    val attributesFromDynamo: Future[(String, Option[Attributes])] = dynamoAttributeService.get(identityId) map { attributes =>
+    val attributesFromDynamo: Future[Option[DynamoAttributes]] = dynamoAttributeService.get(identityId) map { attributes =>
       dynamoAttributeService.ttlAgeCheck(attributes, identityId, forDate)
-      ("Dynamo", attributes)
+      attributes
     }
 
-    val attributesFromZuora: Future[Disjunction[String, Option[Attributes]]] = attributesDisjunction.run.map {
+    val defaultIfZuoraFails: Future[Option[ZuoraAttributes]] = attributesFromDynamo map { maybeDynamoAttributes => maybeDynamoAttributes map {DynamoAttributes.asZuoraAttributes(_)}}
+
+    val attributesFromZuora: Future[Option[ZuoraAttributes]] = defaultIfZuoraFails flatMap  { default =>
```

Comment:
maybe we could call this fallback, rather than default?

Reply:
Yep, will do.